polaris-migrator/src/migrations/replace-border-declarations/replace-border-declarations.ts

```diff
@@ -0,0 +1,307 @@
+
+      handleBorderProps();
+
+      if (targets.some(({replaced}) => !replaced || hasNumericOperator)) {
+        // Insert comment if the declaration value contains calculations
+        decl.before(
+          createInlineComment(POLARIS_MIGRATOR_COMMENT, {prose: true}),
+        );
+        decl.before(
+          createInlineComment(`${decl.prop}: ${parsedValue.toString()};`),
+        );
+      } else {
+        decl.value = parsedValue.toString();
+      }
+
+      //
+      // Handlers
+      //
+
+      function handleBorderProps() {
+        parsedValue.walk((node) => {
+          if (node.type === 'word') {
+            if (globalValues.has(node.value)) return;
+            if (isNumericOperator(node)) {
+              hasNumericOperator = true;
+              return;
+            }
+
+            const dimension = valueParser.unit(node.value);
+
+            if (!isTransformableLength(dimension)) return;
+
+            targets.push({replaced: false});
+
+            const valueInPx = toTransformablePx(node.value);
+
+            if (decl.prop.includes('radius')) {
+              if (!isKeyOf(borderRadiusLengthMap, valueInPx)) {
+                return;
+              }
+
+              node.value = `var(${borderRadiusLengthMap[valueInPx]})`;
+            } else {
+              if (!isKeyOf(borderWidthLengthMap, valueInPx)) {
+                return;
+              }
+
+              node.value = `var(${borderWidthLengthMap[valueInPx]})`;
+            }
+
+            targets.at(-1)!.replaced = true;
+
+            return;
+          }
+
+          if (node.type === 'function') {
+            if (isSassFunction(namespacedRem, node)) {
+              targets.push({replaced: false});
+
+              const args = getFunctionArgs(node);
+
+              if (args.length !== 1) return;
+
+              const valueInPx = toTransformablePx(args[0]);
+
+              if (decl.prop.includes('radius')) {
+                if (!isKeyOf(borderRadiusLengthMap, valueInPx)) {
+                  return;
+                }
+
+                node.value = 'var';
+                node.nodes = [
+                  {
+                    type: 'word',
+                    value: borderRadiusLengthMap[valueInPx],
+                    sourceIndex: node.nodes[0]?.sourceIndex ?? 0,
+                    sourceEndIndex: borderRadiusLengthMap[valueInPx].length,
+                  },
+                ];
+              } else {
+                if (!isKeyOf(borderWidthLengthMap, valueInPx)) {
+                  return;
+                }
+
+                node.value = 'var';
+                node.nodes = [
+                  {
+                    type: 'word',
+                    value: borderWidthLengthMap[valueInPx],
+                    sourceIndex: node.nodes[0]?.sourceIndex ?? 0,
+                    sourceEndIndex: borderWidthLengthMap[valueInPx].length,
+                  },
+                ];
+              }
```

Comment:
This covers `rem()` mapping for both `--p-border-width-X` and `--p-border-radius-X` tokens.